### DISCUSSION TITLE
non-square image issue

### DISCUSSION THREAD

**Discussion by ghost** [nothings/stb#1221] [Q&A]

**Comment by nothings** [comment]
most likely it's because on load, 'x' is the number of bytes per pixel, but you assume it's 3. to force it to be 3, pass in '3' instead of '0' as the last parameter to load.

this doesn't actually fit the described issue, but I don't see anything else obviously wrong offhand, so worth checking.

**Reply by nothings** [reply]
Another thing to consider is writing it out as png or bmp rather than jpg, since writing out as jpeg is lossy. Again, shouldn't matter for the specific issue you're having, but it's an odd choice if the purpose of writing it out is just to debug it.

**Reply by nothings** [reply]
Please read my first reply. You're now allocating 3 bytes per pixel and copying 4 bytes per pixel and totally ignoring how many bytes-per-pixel the image actually loads with.
